### DISCUSSION TITLE
모바일 기기에서 그림자 질문 :(

### DISCUSSION THREAD

**Discussion by kimyang-sun** [naverz/zepeto-studio-kor#1270] [5. Q&A]
안녕하세요 매번 빠른 답변 감사드립니다. :)

현재 기존 에디터에서는

![image](https://user-images.githubusercontent.com/56675004/188824695-b0ad016a-a0b5-4681-badc-351758e7e8d1.png)

이렇게 그림자가 잘 출력이 되는 상태입니다.

하지만 테스트 링크를 통해 앱에서 테스트할때는

![image](https://user-images.githubusercontent.com/56675004/188824852-b045f0c2-d2c7-472e-8814-104f2ab1f7b5.png)

이렇게 그림자가 없는 상태로 출력이 되는 상태입니다.

깃허브 질문들을 찾아봤을때 프로젝트 세팅에서 퀄리티를 조정해줘야 한다는 내용을 보고

![image](https://user-images.githubusercontent.com/56675004/188825202-43efb7f4-3c86-4832-b29e-e5f62ea06fe9.png)

이렇게 모바일 기기에서도 울트라로 변경까지 해둔 상태인데도 똑같이 그림자가 출력이 되지 않습니다.

다른 설정이 필요한걸까요???

혹시 씬 자체 메인 light의 문제일까 싶어 전체 light 인스펙터도 첨부합니다 ㅠ.ㅠ

![image](https://user-images.githubusercontent.com/56675004/188825411-c137132f-7f93-4d43-a99a-a4f0b99c7577.png)

**Comment by chlguswls32** [comment]
안녕하세요 kimayang-sun님
프로젝트 세팅의 Quality부분은 기본값같아보이네요. 따로 세팅하실것은없을것같습니다.

3번째이미지의 Directional Light는 기본값에서 조금 조정된 수치값들이 보이는데, 
ShadowType아래의 Strength, Bias, Normal Bias, Near Plane의 수치를 만져보시면 좋을것같습니다.

일반적으로 그림자가 나올만한 수치값도 안내드립니다.
Strength = 0.7, Bias = 0.05, Normal Bias = 0, Near Plane = 0.2

**Reply by kimyang-sun** [reply]
@chlguswls32  답변 감사합니다!!
수치적용도 해보고 아래 답변두 확인해봤는데 
전체적인 밝기만 많이 밝아지고 그림자는 생기지 않네요 ㅠ.ㅠ

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 윗분이 답변해주신 것처럼 태양 역할을 할 Directional Light이 올바르게 들어가 있고 그림자 세팅값이 되어 있다면
모바일에서도 그림자가 보여져야 할 것입니다. 
<img width="759" alt="Screen Shot 2022-09-08 at 3 38 43 PM" src="https://user-images.githubusercontent.com/100118650/189052356-612fbf94-9794-4b60-8d39-b2ac6a92fbda.png">
![IMG_5843](https://user-images.githubusercontent.com/100118650/189052837-c3cc93ae-be74-4925-8014-eb09d3de95bd.jpg)

아래 유니티의 그림자 설정 트러블슈팅 문서 참고 부탁 드립니다. 
https://docs.unity3d.com/kr/current/Manual/ShadowPerformance.html

**Reply by chlguswls32** [reply]
이미지를 봐서는 DirectionalLight의 빛의 세기도 적용이안되어서 위의 이미지보다, 아래이미지가 어두워보이는것같아요. Shadow에대한 이슈 해결을 먼저 시도해보시기 보다, DirectionalLight가 테스트시 적용되는것인지 확인해보시면 좋을것같아요 ㅎㅎ.

간단하게는 DirecionalLight의 Color를 red로 바꾸고 테스트씬에서 붉게 빛이 적용되었는지 보시면 간단할것같네요 ㅎㅎ 파이팅입니다.

**Reply by kimyang-sun** [reply]
@Zepeto-TechEvangelist 
@chlguswls32 
다들 답변 정말 감사합니다!
다시 제대로 확인해서 테스트 해볼게요 감사합니다 ㅎㅎ

**Comment by chlguswls32** [comment]
Play후에 유니티 에디터의 Hierarchy에서 DirectionalLight가 1개만 잡히는지도 확인해보면 좋을것같습니다.

Directional Light가 여러개일때 여러 우선순위 찾는 방법이 유니티내부적으로 있겠지만,
일반적으로 Intensity가 높은것을 Main DirectionalLight로 잡기때문에 확인이 필요해보입니다.

원인파악을 위한 조금더 전문적인 방법으로는

Window - Analysis - FrameDebugger에서 파악이 가능한데, 보는 방법을 조금 익히셔야 할수있습니다. 
도움되실것같아서 말씀드립니다, 좋은하루되세요.
